pyomo/contrib/mindtpy/feasibility_pump.py

```diff
@@ -0,0 +1,213 @@
+from pyomo.core import (Var, Objective, Reals, minimize,
+                        RangeSet, Constraint, Block, sqrt, TransformationFactory, ComponentMap, value)
+from pyomo.opt import SolverFactory, SolutionStatus
+from pyomo.contrib.gdpopt.util import SuppressInfeasibleWarning, _DoNothing, get_main_elapsed_time, copy_var_list_values, is_feasible
+from pyomo.contrib.mindtpy.nlp_solve import (solve_NLP_subproblem,
+                                             handle_NLP_subproblem_optimal, handle_NLP_subproblem_infeasible,
+                                             handle_NLP_subproblem_other_termination)
+from pyomo.contrib.mindtpy.cut_generation import add_oa_cuts, add_nogood_cuts
+from pyomo.opt import TerminationCondition as tc
+from pyomo.contrib.mindtpy.util import generate_Norm2sq_objective_function
+from pyomo.contrib.mindtpy.mip_solve import solve_MIP_master, handle_master_mip_optimal
+from pyomo.util.infeasible import log_infeasible_constraints
+
+
+def feas_pump_converged(solve_data, config, discrete_only=True):
+    """Calculates the euclidean norm between the discretes in the mip and nlp models"""
+    distance = (sum((nlp_var.value - milp_var.value)**2
+                    for (nlp_var, milp_var) in
+                    zip(solve_data.working_model.MindtPy_utils.variable_list,
+                        solve_data.mip.MindtPy_utils.variable_list)
+                    if (not discrete_only) or milp_var.is_integer()))
+
+    return distance <= config.integer_tolerance
```

Comment:
I think that this convergence is a little harder than it should be. Since you are comparing the integer tolerance of a single variable against the quadratic sum of all discrete variables. Should we compare it against the maximum difference? Or the average? What do you think?